src/jit/lsra.cpp

```diff
@@ -3595,16 +3597,16 @@ regNumber LinearScan::allocateBusyReg(Interval* current, RefPosition* refPositio
                     // codegen considers them as contained memory operands.
                     CLANG_FORMAT_COMMENT_ANCHOR;
 #ifdef _TARGET_ARM_
-                    // TODO-CQ-ARM: Just conservatively "and" two condition. We may implement better condision later.
+                    // TODO-CQ-ARM: Just conservatively "and" two conditions. We may implement a better condition later.
                     isBetterLocation = true;
                     if (recentAssignedRef != nullptr)
-                        isBetterLocation &= (recentAssignedRef->reload && recentAssignedRef->AllocateIfProfitable());
+                        isBetterLocation &= (recentAssignedRef->reload && recentAssignedRef->RegOptional());
 
                     if (recentAssignedRef2 != nullptr)
-                        isBetterLocation &= (recentAssignedRef2->reload && recentAssignedRef2->AllocateIfProfitable());
+                        isBetterLocation &= (recentAssignedRef2->reload && recentAssignedRef2->RegOptional());
 #else
-                    isBetterLocation = (recentAssignedRef != nullptr) && recentAssignedRef->reload &&
-                                       recentAssignedRef->AllocateIfProfitable();
+                    isBetterLocation =
+                        (recentAssignedRef != nullptr) && recentAssignedRef->reload && recentAssignedRef->RegOptional();
```

Comment:
I'm not crazy about the use of `&` for boolean conditions here, but this is just a renaming, and I decided against further cleanup here.

Reply:
I don't understand, what's wrong with `&` (actually `&&` in code)?

Reply:
Sorry for not being clearer - my issue is with the `&=` conditions above. I don't like the mix of booleans with bit-wise operators. While it's correct (at least in this case), it's not clean IMO, and can lead to bugs. when mixed too freely.

Reply:
Ah, sorry, didn't notice the `&=` above. I don't know, it seems fine to me. It's just a bit confusing that on ifdef does it one way and the other does it a different way.